### ISSUE TITLE
Specify flask >=1.0, <1.1.1?

### ISSUE THREAD

**Issue by escherba** [conda-forge/airflow-feedstock#32]
Issue:

It seems to me that Conda feedstock should specify `flask >=1.0, <1.1.1` instead of `<2.0`, otherwise conda will install flask 1.1.1 which expects an incompatible (later) version of werkzeug, which results in an error:
```
>>> import airflow
...
ModuleNotFoundError: No module named 'werkzeug.wrappers.json'; 'werkzeug.wrappers' is not a package
```

My current workaround is installing airflow like this:
```
conda install -c conda-forge airflow flask=1.0.3
```

Alternatively, one could drop the strict werkzeug requirement (though that could break things in Airflow; I haven't verified if that would work yet).

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
airflow                   1.10.3post1              py36_0    conda-forge
alembic                   0.9.9                      py_0    conda-forge
asn1crypto                0.24.0                py36_1003    conda-forge
babel                     2.7.0                      py_0    conda-forge
ca-certificates           2019.6.16            hecc5488_0    conda-forge
certifi                   2019.6.16                py36_1    conda-forge
cffi                      1.12.3           py36h8022711_0    conda-forge
chardet                   3.0.4                 py36_1003    conda-forge
click                     7.0                        py_0    conda-forge
colorama                  0.4.1                      py_0    conda-forge
configparser              3.5.0                 py36_1001    conda-forge
croniter                  0.3.30                     py_0    conda-forge
cryptography              2.7              py36h72c5cf5_0    conda-forge
defusedxml                0.5.0                      py_1    conda-forge
dill                      0.2.9                    py36_0    conda-forge
flask                     1.0.3                      py_0    conda-forge
flask-admin               1.5.3                      py_0    conda-forge
flask-appbuilder          1.12.3                     py_0    conda-forge
flask-babel               0.12.2                     py_0    conda-forge
flask-caching             1.3.3                      py_0    conda-forge
flask-login               0.4.1                      py_0    conda-forge
flask-openid              1.2.5                 py36_1003    conda-forge
flask-sqlalchemy          2.4.0                      py_0    conda-forge
flask-swagger             0.2.13                     py_2    conda-forge
flask-wtf                 0.14.2                     py_1    conda-forge
future                    0.16.0                py36_1002    conda-forge
gitdb2                    2.0.5                      py_0    conda-forge
gitpython                 2.1.12                     py_0    conda-forge
gunicorn                  19.9.0                py36_1000    conda-forge
icu                       64.2                 he1b5a44_0    conda-forge
idna                      2.8                   py36_1000    conda-forge
iso8601                   0.1.12                     py_1    conda-forge
itsdangerous              1.1.0                      py_0    conda-forge
jinja2                    2.10                       py_1    conda-forge
json-merge-patch          0.2                        py_0    conda-forge
libblas                   3.8.0               10_openblas    conda-forge
libcblas                  3.8.0               10_openblas    conda-forge
libedit                   3.1.20181209         hc058e9b_0  
libffi                    3.2.1                hd88cf55_4  
libgcc-ng                 9.1.0                hdf63c60_0  
libgfortran-ng            7.3.0                hdf63c60_0  
libiconv                  1.15              h516909a_1005    conda-forge
liblapack                 3.8.0               10_openblas    conda-forge
libopenblas               0.3.6                h6e990d7_4    conda-forge
libstdcxx-ng              9.1.0                hdf63c60_0  
libxml2                   2.9.9                hee79883_2    conda-forge
libxslt                   1.1.32            h31b3aaa_1004    conda-forge
lockfile                  0.12.2                     py_1    conda-forge
lxml                      4.3.4            py36h7ec2d77_0    conda-forge
mako                      1.0.10                     py_0    conda-forge
markdown                  2.6.11                     py_0    conda-forge
markupsafe                1.1.1            py36h14c3975_0    conda-forge
ncurses                   6.1                  he6710b0_1  
numpy                     1.16.4           py36h95a1406_0    conda-forge
openblas                  0.3.6                h6e990d7_4    conda-forge
openssl                   1.1.1c               h516909a_0    conda-forge
pandas                    0.25.0           py36hb3f55d8_0    conda-forge
pendulum                  1.4.4                    py36_1    conda-forge
pip                       19.1.1                   py36_0  
psutil                    5.6.3            py36h516909a_0    conda-forge
pycparser                 2.19                     py36_1    conda-forge
pygments                  2.4.2                      py_0    conda-forge
pyopenssl                 19.0.0                   py36_0    conda-forge
pysocks                   1.7.0                    py36_0    conda-forge
python                    3.6.8                h0371630_0  
python-daemon             2.1.2                    py36_0    conda-forge
python-dateutil           2.8.0                      py_0    conda-forge
python-editor             1.0.4                      py_0    conda-forge
python-nvd3               0.15.0                     py_1    conda-forge
python-slugify            1.2.6                      py_0    conda-forge
python3-openid            3.1.0                 py36_1001    conda-forge
pytz                      2019.1                     py_0    conda-forge
pytzdata                  2019.2                     py_0    conda-forge
pyyaml                    5.1.1            py36h516909a_0    conda-forge
readline                  7.0                  h7b6447c_5  
requests                  2.22.0                   py36_1    conda-forge
setproctitle              1.1.10          py36h14c3975_1001    conda-forge
setuptools                41.0.1                   py36_0  
six                       1.12.0                py36_1000    conda-forge
smmap2                    2.0.5                      py_0    conda-forge
sqlalchemy                1.2.18           py36h14c3975_0    conda-forge
sqlite                    3.29.0               h7b6447c_0  
tabulate                  0.8.3                      py_0    conda-forge
tenacity                  4.12.0                   py36_1    conda-forge
text-unidecode            1.2                        py_0    conda-forge
thrift                    0.11.0          py36hf484d3e_1001    conda-forge
tk                        8.6.8                hbc83047_0  
tzlocal                   2.0.0                      py_0    conda-forge
unicodecsv                0.14.1                     py_1    conda-forge
unidecode                 1.1.1                      py_0    conda-forge
urllib3                   1.25.3                   py36_0    conda-forge
werkzeug                  0.14.1                     py_0    conda-forge
wheel                     0.33.4                   py36_0  
wtforms                   2.2.1                      py_0    conda-forge
xz                        5.2.4                h14c3975_4  
yaml                      0.1.7             h14c3975_1001    conda-forge
zlib                      1.2.11               h7b6447c_3  
zope.deprecation          4.4.0                      py_0    conda-forge

```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

     active environment : airflow
    active env location : /home/eugene/anaconda3/envs/airflow
            shell level : 2
       user config file : /home/eugene/.condarc
 populated config files : 
          conda version : 4.7.10
    conda-build version : 3.18.7
         python version : 3.7.3.final.0
       virtual packages : __cuda=10.1
       base environment : /home/eugene/anaconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/eugene/anaconda3/pkgs
                          /home/eugene/.conda/pkgs
       envs directories : /home/eugene/anaconda3/envs
                          /home/eugene/.conda/envs
               platform : linux-64
             user-agent : conda/4.7.10 requests/2.21.0 CPython/3.7.3 Linux/4.18.0-25-generic ubuntu/18.04.2 glibc/2.27
                UID:GID : 1001:1001
             netrc file : None
           offline mode : False

```
</details>

**Comment by xylar** [comment]
@escherba, thanks for bringing this issue to our attention.  I think this might suggest that the problem is with the flask feedstock (for flask 1.1.1) and not with `airflow` itself.  If I just do

```
conda create -y -n test python=3.7 airflow flask
conda activate test
python -c "import flask"
```
I get flask 1.1.1 and werkzeug 0.14.1 as you do and see the same error as above.  Could you please open an issue with flask-feedstock and refer to this issue?

**Comment by xylar** [comment]
Regarding package constraints, I've been following the main repo's constraints as closely as I can and I can modify them but would prefer not to if we can find another solution:
https://github.com/apache/airflow/blob/1.10.3post2/setup.py

**Comment by xylar** [comment]
Okay, I've had a chance to look into this a bit more.  flask needs to be fixed, and I've issued a PR.  This would have handled the dependency.  But I also agree that there's not going to be a version of flask>=1.1.1 that works with the current constraint on werkzeug so we might as well update the dependency.  I'll issue a PR for that here.

**Comment by xylar** [comment]
@escherba, one more update.  I tried to rebuild with the constraint and it didn't work because apparently airflow requires werkzeug>=0.15.0, independent of flask and the specs are simply incorrect.  I just merged a PR (#33).  Once it's on the `conda-forge` channel, I'll retest this issue and reopen if it's not fixed.

**Comment by xylar** [comment]
I was able to successfully run:
```
conda create -y -n test python=3.7 airflow flask
conda activate test
python -c "import flask"
```
It installed `airflow=1.10.3post2`.  @escherba, please make sure the same is true for you.
